src/main/java/net/minecraftforge/common/util/EnumHelper.java

```diff
@@ -141,6 +143,19 @@ public static EnumRarity addRarity(String name, TextFormatting color, String dis
     {
         return addEnum(EnumRarity.class, name, color, displayName);
     }
+    /**
+     * 
+     * @param name - The name the new enum constant should have
+     * @param textureLocation - The path to the texture for this armor type. Formatted as domain:path.
+     * @param hash - A short unique 3 letter name for the armor type 
```

Comment:
> Is that still relevant here
Currently, yes.

I've just realised that there is a significant issue with the hashes. Each HorseArmorType **must** have a unique hash. If two types have the same hash, the wrong texture will be rendered in certain cases.  

An example:  
There are two horses, A & B, both have the **same base coat and markings**. There are also two horse armors, Y & Z, both have the **same hash**. A wears armor Y, B wears armor Z. The position of the two horses doesn't matter, as long as the player sees them in the same Minecraft session. If A is rendered first, both armors will use armor Y's texture. If B is rendered first, both armors will use armor Z's texture. This is due to MC caching the layered texture for a particular combination of coat, markings and armor in a map with the concatenated hashes as the key.

I can think of two solutions to this:  
1. Prepend the modid to the hash. 
2. Generate hashes that are three letter alphabetic sequences. So the first armor type registered would get the hash 'aaa', the second 'aab', the third 'aac', etc.  This would keep hashcode computation time constant. However it would limit the number of possible horse armor type to 26^3(17,576), that should be more than enough though. The hashes don't need to be persistent as the texture cache isn't persistent.

1 is simple, however I'd prefer 2 as the hash is really an internal implementation detail that a modder doesn't need to deal with. Since we can generate it, I think we should. I also prefer 2 as it keeps the hashes at their current length. I wrote a string hashcode(test program)[https://gist.github.com/Leviathan143/d7c0da865dbbbf11615a2f2357d49d3f] and the hashcode computation times are extremely fast (The data and best/worst case scenarios are included in the linked gist), however I assume the hashes were kept short for some reason and using 2 means we don't have to worry about what that reason is.